src/gnc/attitude_estimator.cpp

```diff
@@ -438,10 +439,35 @@ void attitude_estimator_reset(AttitudeEstimatorState &state,
     double t, lin::Vector3d const &r_ecef, lin::Vector3f const &b_body,
     lin::Vector3f const &s_body) {
   GNC_ASSERT_NORMALIZED(s_body);
-
-  // TODO : Implement this
-  // https://github.com/pathfinder-for-autonomous-navigation/psim/issues/187
+  
   lin::Vector4f q_body_eci = lin::nans<float, 4, 1>();
+  
+  lin::Vector3f b_eci{};
+  lin::Vector4f q{};
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);  // b_eci = b_ecef
```

Comment:
Instead of the comments, why not just add variables that are named things like q_eci_ecef, as part of the code? Then the code is self-documenting.

Reply:
Because then we are growing the stack for no reason. The intention with these comments is to let you know what is in the variable `b_eci` at a given line. By the end of our calculation's we want it to be the magnetic field in an inertial frame.

Reply:
I heavily disagree with trying to save stack space at the cost of making code more readable. We should only do this if stack space starts proving to be a problem.

At the very least, we should modify the variable name from `b_eci` to something else if it's not always the magnetic field vector in ECI.

Reply:
Compilers are very good at eliminating unnecessary stack use so I actually think this concern is unwarranted

Reply:
This doesn't seem like an issue worth holding this PR up especially considering the code _is_ documented with how the local variable is being mutated. I also prefer this method on handling local variables because it prevents polluting the local namespace with a ton of variables that we don't need and is more explicit with the values we are actually looking to calculate - we have code following this style all over the place in GNC land.

Reply:
Besides the stack deal it helps we work with intellisense better to be quite honest.

Reply:
Don't really care though we should just merge this in one way or the other.

Reply:
Shihao is changing this now I think